CMakeLists.txt

```diff
@@ -85,7 +85,7 @@ if(${COMPILER_IS_LIKE_GNU})
     set(SPIRV_WARNINGS ${SPIRV_WARNINGS} -Werror)
   endif()
 elseif(MSVC)
-  set(SPIRV_WARNINGS -D_CRT_SECURE_NO_WARNINGS /wd4800)
+  set(SPIRV_WARNINGS -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS /wd4800)
```

Comment:
Can you explain why you need to disable the warning?  Not in the code, but in a comment in the PR.  I don't know what the warning is, so I have not idea if this is a good idea or not.

Reply:
Windows issues warning C4996 (and we have warnings as errors) for the std::equal I added to compare uint32_t's. It suggests using a version of std::equal that requires c++14 or a Windows specific extension. The warning is due to unchecked iterators (from the pointer increments). Linux and Mac are ok with this code and we only use c++11 so I can't use the newer version of std::equal.

Reply:
I should also mention the other define there "_CRT_SECURE_NO_WARNINGS" generates the same warning number in Visual Studio.